src/main/java/ch/njol/skript/expressions/ExprFuseDuration.java

```diff
@@ -0,0 +1,154 @@
+import org.bukkit.entity.TNTPrimed;
+import org.bukkit.entity.minecart.ExplosiveMinecart;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Fuse Duration")
+@Description({
+	"Sets or gets the fuse duration of an entity.",
+	"Supports Primed TNT, Creepers, and TNT Minecarts.",
+	"For Creepers, you can also retrieve their maximum fuse time."
+})
+@Examples({
+	"set fuse duration of {_tnt} to 4 seconds",
+	"set max fuse duration of last spawned creeper to 3 seconds",
+	"broadcast remaining fuse time of {_tnt}",
+})
+@Since("INSERT VERSION")
+public class ExprFuseDuration extends SimplePropertyExpression<Entity, Timespan> {
+
+	static {
+		register(ExprFuseDuration.class, Timespan.class, "[remaining|:max[imum]] fuse [duration|time]", "entities");
+	}
+
+	private boolean max;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		max = parseResult.hasTag("max");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	public @Nullable Timespan convert(Entity entity) {
+		if (entity instanceof TNTPrimed) {
+			return getTimespan(((TNTPrimed) entity).getFuseTicks());
+		} else if (entity instanceof Creeper) {
+			int fuseTicks = max ? ((Creeper) entity).getMaxFuseTicks() : ((Creeper) entity).getMaxFuseTicks() - ((Creeper) entity).getFuseTicks();
+			return getTimespan(fuseTicks);
+		} else if (entity instanceof ExplosiveMinecart) {
+			return getTimespan(((ExplosiveMinecart) entity).getFuseTicks());
+		}
+		return null;
+	}
+
+	private Timespan getTimespan(int ticks) {
+		return new Timespan(Timespan.TimePeriod.TICK, Math.max(ticks, 0));
+	}
+
+
+	@Override
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		return switch (mode) {
+			case SET, ADD, REMOVE, DELETE, RESET -> CollectionUtils.array(Timespan.class);
+			default -> null;
+		};
+	}
+
+	@Override
+	public void change(Event event, Object @NotNull [] delta, ChangeMode mode) {
+		for (Entity entity : getExpr().getArray(event)) {
+			if (entity instanceof TNTPrimed) {
+				changeTNTFuseTicks((TNTPrimed) entity, delta, mode);
+			} else if (entity instanceof Creeper) {
+				changeCreeperFuseTicks((Creeper) entity, delta, mode);
+			} else if (entity instanceof ExplosiveMinecart) {
+				changeMinecartFuseTicks((ExplosiveMinecart) entity, delta, mode);
+			}
+		}
+	}
+
+	private void changeTNTFuseTicks(TNTPrimed tnt, Object[] delta, ChangeMode mode) {
+		int currentTicks = tnt.getFuseTicks();
+		int newTicks = calculateNewTicks(tnt, currentTicks, delta, mode);
+		tnt.setFuseTicks(newTicks);
+	}
+
+	private void changeCreeperFuseTicks(Creeper creeper, Object[] delta, ChangeMode mode) {
+		if (max) {
+			int currentMaxTicks = creeper.getMaxFuseTicks();
+			int newMaxTicks = calculateNewTicks(creeper, currentMaxTicks, delta, mode);
+			creeper.setMaxFuseTicks(Math.max(newMaxTicks, 0));
```

Comment:
Should there be a return after this? As in, is it intended to run the code after?